config.py

```diff
@@ -226,62 +178,112 @@ def __with_lock_and_backup(self, path: Path, backup: bool, callable: Callable[[]
                     raise Exception(f"Failed to save config: {e}")
         finally:
             # Cleanup lock and backup files:
-            if lock_path.exists():  # Only try to delete if it exists
-                try:
-                    lock_path.unlink()
-                except FileNotFoundError:
-                    pass  # Ignore if file doesn't exist
-            
+            if lock_path.exists():
+                lock_path.unlink()
+
             if backup and backup_path and backup_path.exists():
-                try:
-                    backup_path.unlink()
-                except FileNotFoundError:
-                    pass  # Ignore if file doesn't exist
+                backup_path.unlink()
 
     @abstractmethod
-    def _handle_loaded_config(self, data):
+    def _handle_loaded_config(self, json_string: str):
         pass
 
-class AlltalkNewEnginesConfig(AbstractJsonConfig):
+    def to_dict(self):
+        return self.__delegate.model_dump(by_alias=True)
+
+    # Delegation to the loaded config data

```

Comment:
This is the most complex part dealing with delegation from e.g. `AlltalkNewEnginesConfig` to `AlltalkNewEnginesConfigModel`. I took most of the code from here: https://www.fast.ai/posts/2019-08-06-delegation.html